usr/share/rear/prep/BAREOS/default/500_check_BAREOS_bconsole_results.sh

```diff
@@ -3,39 +3,41 @@
 # are all  the files/directories present?
 
 if [ "$BEXTRACT_DEVICE" -o "$BEXTRACT_VOLUME" ]; then
-
-   ### Bacule restore using bextract, no need for a director
+   ### restore using bextract, no need for a director
    return
 fi
 
-# Check bconsole version and use appropriate CLI switch:
-# Bareos 22 introduced a breaking change how its CLI tools (such as bconsole) parse the arguments.
-# Before it was "bconsole -xc" but since Bareos 22 it must be "bconsole --xc"
-# (otherwise it terminates with exit code 113 so "rear mkrescue" will fail).
-# Before "bconsole ...xc" is called, it now checks the bconsole version
-# and uses the new argument format '--xc' when version >= 22.
-# The backward compatible fallback is "bconsole -xc".
-# See https://github.com/rear/rear/issues/2900
-BCONSOLE_VERSION=$(bconsole --version | awk -F '.' '{print $1}')
-if [ "$BCONSOLE_VERSION" -ge 22 ]; then
-	BCONSOLE_XC="--xc"
-else
-	BCONSOLE_XC="-xc"
-fi
+if [ "$PING" ]; then
+
+    if [ -z $BAREOS_DIRECTOR_ADDRESS ]; then
+        # Check bconsole version and use appropriate CLI switch:
+        # Bareos 22 introduced a breaking change how its CLI tools (such as bconsole) parse the arguments.
+        # Before it was "bconsole -xc" but since Bareos 22 it must be "bconsole --xc"
+        # (otherwise it terminates with exit code 113 so "rear mkrescue" will fail).
+        # "bconsole --version" has also been introduced in Bareos 22,
+        # therefore if this command succeeds, we use "--xc".
+        # The backward compatible fallback is "bconsole -xc".
+        # See https://github.com/rear/rear/issues/2900
+        BCONSOLE_XC="-xc"
+        if bconsole --version >/dev/null 2>&1; then
+            BCONSOLE_XC="--xc"
+        fi
+
+        #
+        # See if we can ping the director
+        #
+        # is the director server present? Fetch address from configuration.
+        BAREOS_DIRECTOR_ADDRESS=$(bconsole "$BCONSOLE_XC" | sed -n -r  's/ *address *= *["](.*)["]/\1/ip')
+        [ "${BAREOS_DIRECTOR_ADDRESS}" ]
+        StopIfError "Failed to get Bareos Director address via bconsole configuration. Please configure BAREOS_DIRECTOR_ADDRESS."
```

Comment:
In general plase avoid the ...IfError functions
because they do not work reliably,
see their description in
usr/share/rear/lib/_input-output-functions.sh
currently starting at
https://github.com/rear/rear/blob/master/usr/share/rear/lib/_input-output-functions.sh#L879